### PR TITLE
Add header component and its functionality

### PR DESCRIPTION
**Pull request by AndresForondaPerf**
### Changes

- Added header component and header model
- Configuration:
  - Logo: Incorporate an image from the asset folder, along with alternative text and a hyperlink.
  - Navigation Items:
    - Allows for the addition of up to six navigation items.
    - Active Page Indicator: Automatically highlights the navigation item corresponding to the current page. It's distinguished by a red background with white text.
    - Hover Effect: Enhances user experience with a dynamic hover effect, resembling the appearance of an active item.
    - Click Event: Disabled on active page item, avoiding the reload of the current page
    
### Screenshots

- Header component (About Us page active link)

image

---

- Hover effect:

hover

---

- Logo Selection:

logo pick

---

- Nav Items Max 6:

max 6

---

- Current Page Nav Item disabled 

link disabled

### CODE REVIEW

File: ui.apps/src/main/content/jcr_root/apps/aem-bootcamp/components/header/header.html [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    data-sly-test.empty="${!navHeader.fileReference}"
+    data-sly-use.template="core/wcm/components/commons/v1/templates.html"
+    data-sly-call="${template.placeholder @ isEmpty=empty}" />
+    <sly data-sly-test="${!empty}">
+        <div class="container responsivegrid">
+            <div  class="container responsivegrid cmp-layoutcontainer--header">
+               <div class="cmp-container">
+                  <div class="aem-Grid cmp-container--layout">
+                     <div  class="image cmp-image--logo aem-GridColumn" style="display: inline-block;">
+                        <div  class="cmp-image" href="#">
```
Comment (valMoncadaS) [inline ui.apps/src/main/content/jcr_root/apps/aem-bootcamp/components/header/header.html]: remove href from div tag

**Review by valMoncadaS — CHANGES_REQUESTED** [review]

File: ui.frontend/src/main/webpack/components/_header.scss [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    }
+
+    .cmp-navigation__item:hover .cmp-navigation__item-link {
+      border-bottom: $color-primary 2px solid;
+    }
+
+  
+  }
+
+    // .cmp-container--layout {
```
Comment (valMoncadaS) [inline ui.frontend/src/main/webpack/components/_header.scss]: Please removed commented code

File: ui.frontend/src/main/webpack/components/_helloworld.scss [hunk 3]
```diff
@@ -1,7 +1,9 @@
 // Best practice:
 // For a good separation of concerns, don't rely on the DOM element names,
 // but on dedicated CSS selectors to identify all elements
-.cmp-helloworld {}
+.cmp-helloworld {
```
Comment (valMoncadaS) [inline ui.frontend/src/main/webpack/components/_helloworld.scss]: this is out of the scope of the story.
Please clarify why this is necessary

**Review by valMoncadaS — CHANGES_REQUESTED** [review]

File: ui.frontend/src/main/webpack/components/_header.scss (hunk 2 above)
Comment (AndresForondaPerf) [reply ui.frontend/src/main/webpack/components/_header.scss]: Removed

File: ui.frontend/src/main/webpack/components/_helloworld.scss (hunk 3 above)
Comment (AndresForondaPerf) [reply ui.frontend/src/main/webpack/components/_helloworld.scss]: Changes removed

**Review by valMoncadaS — APPROVED** [review]
Looks good for me. Approved!

**Pull request merged**